src/main/java/io/elastic/jdbc/TableNameProvider.java

```diff
@@ -1,84 +1,88 @@
 package io.elastic.jdbc;
 
-import com.google.gson.JsonObject;
 import io.elastic.api.SelectModelProvider;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import java.sql.Connection;
 import java.sql.DatabaseMetaData;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.util.Arrays;
 import java.util.List;
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class TableNameProvider implements SelectModelProvider {
 
-    private static final Logger logger = LoggerFactory.getLogger(TableNameProvider.class);
+  private static final Logger logger = LoggerFactory.getLogger(TableNameProvider.class);
 
-    public JsonObject getSelectModel(JsonObject configuration) {
-        logger.info("About to retrieve table name");
+  @Override
+  public JsonObject getSelectModel(JsonObject configuration) {
+    logger.info("About to retrieve table name");
 
-        JsonObject result = new JsonObject();
-        Connection connection = null;
-        ResultSet rs = null;
+    JsonObjectBuilder result = Json.createObjectBuilder();
+    Connection connection = null;
+    ResultSet rs = null;
 
-        try {
-            connection = Utils.getConnection(configuration);
-            logger.info("Successfully connected to DB");
+    try {
+      connection = Utils.getConnection(configuration);
+      logger.info("Successfully connected to DB");
 
-            // get metadata
-            DatabaseMetaData md = connection.getMetaData();
+      // get metadata
+      DatabaseMetaData md = connection.getMetaData();
 
-            // get table names
-            String[] types = {"TABLE", "VIEW"};
-            rs = md.getTables(null, "%", "%", types);
+      // get table names
+      String[] types = {"TABLE", "VIEW"};
+      rs = md.getTables(null, "%", "%", types);
 
-            // put table names to result
-            String tableName;
-            String schemaName;
-            Boolean isEmpty = true;
+      // put table names to result
+      String tableName;
+      String schemaName;
+      boolean isEmpty = true;
 
-            while (rs.next()) {
-                tableName = rs.getString("TABLE_NAME");
-                schemaName = rs.getString("TABLE_SCHEM");
-                if (configuration.get("dbEngine").getAsString().toLowerCase().equals("oracle")
-                        && isOracleServiceSchema(schemaName)) {
-                    continue;
-                }
-                if (schemaName != null) {
-                    tableName = schemaName + "." + tableName;
-                }
-                result.addProperty(tableName, tableName);
-                isEmpty = false;
-            }
-            if (isEmpty) {
-                result.addProperty("", "no tables");
-            }
+      while (rs.next()) {
+        tableName = rs.getString("TABLE_NAME");
+        schemaName = rs.getString("TABLE_SCHEM");
```

Comment:
it is normal "TABLE_SCHEM" ??? May be "TABLE_SCHEM**A**" ??

Reply:
no need changes (was discussed)